internal/backends/sqlite/metadata/pool/pool.go

```diff
@@ -50,17 +51,17 @@ type Pool struct {
 }
 
 // New creates a pool for SQLite databases in the given directory.
-func New(dir string, l *zap.Logger) (*Pool, error) {
+func New(uri *url.URL, l *zap.Logger) (*Pool, error) {
 	// TODO accept URI with a directory name, not just directory name
 	// https://github.com/FerretDB/FerretDB/issues/2753
 
-	matches, err := filepath.Glob(filepath.Join(dir, "*"+filenameExtension))
+	matches, err := filepath.Glob(filepath.Join(uri.Path, "*"+filenameExtension))
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
 	p := &Pool{
-		dir:   dir,
+		dir:   uri.Path,
```

Comment:
🛑 And after that, we lose query parameters that we should pass to `openDB` below, as well in `GetOrCreate` via `databasePath` and maybe somewhere else

Reply:
Saved `uri` instead of `dir`.

Reply:
That does not change the fact that openDB accepts matched files (without query parameters) in one place and the database path (also without query parameters) in another.